### PR TITLE
AP_Motors: MatrixTS default to normal Matrix 

### PR DESCRIPTION
**Pull request by IamPete1**
This work was funded by FDrones.

There are two things in this PR. Firstly it removes matrixTS frame classes and types that are the same as the default matrix and then uses the default matrix setup. This also allows any frame type supported by copter to be used as a tailsitter. The recently added specialist no yaw frame types are retained. 

Secondly it removes the output_motor_mask function in matrixTS in favor of the [default one](https://github.com/ArduPilot/ardupilot/blob/040a1b7fbeec726fb4696a669f2c99eb85969076/libraries/AP_Motors/AP_MotorsMulticopter.cpp#L733). The default one has slew rates and thrust linearization that this one was missing. Maybe there was some reason not to have these? I have added the efficiency improvements to the default one.

### CODE REVIEW

**Comment by IamPete1** [comment]
I have also removed output to motors in matrix TS. We don't need to set the roll, pitch and yaw as we fetch them directly from motors in tailsitter.cpp. 

Looking at this more I suspect we could actually remove MatrixTS altogether and put the no yaw frames in Matrix now they have there own types. I think the only difference in output_armed_stabilizing() is that the yaw stuff is removed. We could auto set the yaw head room to zero for the no yaw frames and I think this would have the same effect?

**Comment by kd0aij** [comment]
Have you tested the change to output_motor_mask? 
It looks like it undoes a regression fix that you suggested and I submitted (many moons ago): https://github.com/ArduPilot/ardupilot/pull/10354

**Comment by IamPete1** [comment]
Yeah, it does still work, I don't think it undoes the patch in that PR

**Comment by kd0aij** [comment]
Right, that PR changed AP_MotorsMulticopter, not AP_MotorsMatrixTS
I don't suppose it makes a significant difference, but the thrust linearization and slew rate stuff is pretty specific to multicopters.
The other changes look pretty safe to me; just eliminating code redundancy.
@rmackay9  may change his mind on moving tailsitter support into MotorsMatrix now...

File: libraries/AP_Motors/AP_MotorsMulticopter.cpp [hunk 1]
```diff
@@ -732,6 +732,9 @@ void AP_MotorsMulticopter::set_throttle_passthrough_for_esc_calibration(float th
 // the range 0 to 1
 void AP_MotorsMulticopter::output_motor_mask(float thrust, uint8_t mask, float rudder_dt)
 {
+    const int16_t pwm_min = get_pwm_output_min();
```
Comment (tridge) [inline libraries/AP_Motors/AP_MotorsMulticopter.cpp]: why are these signed?

**Pull request merged**